### PR TITLE
CPU optimizations in phases 1 and 3

### PR DESCRIPTION
**Pull request by rostislav**

### CODE REVIEW

File: src/phase3.hpp [hunk 1]
```diff
@@ -59,8 +59,8 @@ void WriteParkToFile(
     uint64_t writer = table_start + park_index * park_size_bytes;
     uint8_t *index = park_buffer;
 
-    Bits first_line_point_bits(first_line_point, 2 * k);
-    first_line_point_bits.ToBytes(index);
+    first_line_point <<= 128 - 2 * k;
+    Util::IntTo16Bytes(index, first_line_point);
```
Comment (arvidn) [inline src/phase3.hpp:63]: does the `park_buffer` also have the additional 7 trailing bytes?

File: src/phase3.hpp [hunk 2]
```diff
@@ -399,10 +400,12 @@ Phase3Results RunPhase3(
...
 
             // Write the new position (index) and the sort key
-            Bits to_write = Bits(sort_key, right_sort_key_size);
-            to_write += Bits(index, k + 1);
+            uint128_t to_write = (uint128_t)sort_key << sort_key_shift;
+            to_write |= (uint128_t)index << index_shift;
 
-            L_sort_manager->AddToCache(to_write);
+            uint8_t bytes[16];
+            Util::IntTo16Bytes(bytes, to_write);
```
Comment (arvidn) [inline src/phase3.hpp:407]: this is safe without having the extra 7 bytes because it's all already aligned, right?

**Review by arvidn — APPROVED** [review]

File: src/util.hpp [hunk 3]
```diff
@@ -339,6 +346,40 @@ namespace Util {
         double b = ldexp(frac, exp);
         return b;
     }
+
+#if defined(_WIN32) || defined(__x86_64__)
+    void CpuID(uint32_t leaf, uint32_t regs[4])
```
Comment (arvidn) [inline src/util.hpp:351]: passing an array as a parameter is misleading. this is the same as `uint32_t*`. i.e. the 4 is ignored.
If you want to enforce passing a C-array of 4, you can take a reference to the array, or use `std::array<uint32_t, 4>`

**Review by arvidn — APPROVED** [review]
just one nit

File: src/util.hpp (hunk 3 above)
Comment (rostislav) [reply src/util.hpp:351]: Thanks, I'll change it to `uint32_t *`.

**Pull request merged**